drivers/nrf_radio_802154/include/nrf_802154.h

```diff
@@ -0,0 +1,1292 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY, AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @defgroup nrf_802154 802.15.4 Radio Controller
+ * @{
+ *
+ */
+
+#ifndef NRF_802154_H_
+#define NRF_802154_H_
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "nrf_802154_types.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define NRF_802154_VER_MAJ 2  // !< Major version of the 802.15.4 Radio Controller
+#define NRF_802154_VER_MIN 0  // !< Minor version of the 802.15.4 Radio Controller
+#define NRF_802154_VER_PAT 0  // !< Patch version of the 802.15.4 Radio Controller
+
+/** @brief Macro containing combined version of the 802.15.4 Radio Controller.
+ *
+ * This macro combines major version (as number of hundred of hundreds) with minor version (as
+ * number of hundreds) with patch version (as number) summed together.
+ *
+ * The version 2.0.0 is represented as 20000.
+ * The version 2.0.4 is represented as 20004.
+ * The version 2.14.0 is represented as 21400.
+ */
+#define NRF_802154_VER ((NRF_802154_VER_MAJ * 10000) +                                             \
+                        (NRF_802154_VER_MIN * 100) +                                               \
+                        (NRF_802154_VER_PAT))
+
+/**
+ * @brief Structure with initialization parameters of the module.
+ */
+typedef struct
+{
+    // Intentionally empty
+} nrf_802154_init_cfg_t;
+
+#define NRF_802154_INIT_CFG_DEF (nrf_802154_init_cfg_t){                                           \
+}
+
+/**
+ * @brief Initialize the 802.15.4 driver.
+ *
+ * This function initializes the RADIO peripheral in the @ref RADIO_STATE_SLEEP state.
+ *
+ * This function is to be called once, before any other functions from this module. Calling this
+ * function again without calling @ref nrf_802154_uninit earlier results in undefined behavior.
+ *
+ * To reinitialize the module with different configuration call @ref nrf_802154_uninit and
+ * @ref nrf_802154_init again with new configuration.
+ *
+ * @param[in] p_cfg  Structure with initialization parameters of the module. Default initialization
+ *                   parameters are available with @ref NRF_802154_INIT_CFG_DEF macro.
+ */
+void nrf_802154_init(const nrf_802154_init_cfg_t * p_cfg);
```

Comment:
Why is the `include/` directory added? The files it contains don't seem to be used or even to be consistent with the other source files, see for instance the above declaration of `nrf_802154_init()`.

Reply:
removed, those extra files and dir